mobile/lib/components/ra_player/ra_player_widget.dart

```diff
@@ -0,0 +1,314 @@
+        child: ValueListenableBuilder<ProgressBarState>(
+          valueListenable: audioHandler.progress,
+          builder: (context, progress, _) {
+            return Padding(
+              padding: const EdgeInsets.symmetric(
+                horizontal: RaPageConstraints.pagePaddingValue,
+              ),
+              child: ProgressBar(
+                progress: progress.current,
+                total: progress.total,
+                buffered: progress.buffered,
+                thumbColor: context.colors.highlightRed,
+                thumbRadius: RaPlayerWidget._thumbRadius,
+                thumbGlowRadius: RaPlayerWidget._thumbGlowRadius,
+                bufferedBarColor: context.colors.backgroundLightSecondary,
+                baseBarColor: context.colors.backgroundLight,
+                progressBarColor: context.colors.highlightGreen,
+                thumbCanPaintOutsideBar: false,
+                timeLabelLocation: TimeLabelLocation.none,
+                onSeek: audioHandler.seek,
+              ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
+
+class _BackButton extends StatelessWidget {
+  const _BackButton({required this.audioHandler});
+
+  final RaPlayerHandler audioHandler;
+
+  double _calculateButtonWidth(BuildContext context) {
+    final screenWidth = MediaQuery.of(context).size.width;
+    const padding = RaPageConstraints.pagePaddingValue * 2;
+    return (screenWidth - padding) / 3;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: _calculateButtonWidth(context),
+      height: RaPageConstraints.radioPlayerHeight / 2,
+      color: context.colors.backgroundDark,
+      child: GestureDetector(
+        onTap: () async => audioHandler.playMediaItem(radioMediaItem),
+        child: Center(
+          child: Text(
+            context.l10n.backToRadio,
+            style: context.textStyles.textSmallGreen,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _Player extends StatelessWidget {
+  const _Player({required this.audioHandler});
+
+  final RaPlayerHandler audioHandler;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      height: RaPageConstraints.radioPlayerHeight,
+      width: MediaQuery.of(context).size.width,
+      color: context.colors.backgroundDarkSecondary,
+      child: Row(
+        children: [
+          PlayerPlayButton(
+            audioHandler: audioHandler,
+            size: RaPlayerWidget._playerSize,
+          ),
+          _PlayerTitle(
+            audioHandler: audioHandler,
+            width: MediaQuery.of(context).size.width / 1.4,
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+/// The [RaPlayButton] controlling the radio stream.
+class PlayerPlayButton extends StatelessWidget {
+  const PlayerPlayButton({
+    super.key,
+    required this.audioHandler,
+    required this.size,
+    this.padding = const EdgeInsets.symmetric(horizontal: 14),
```

Comment:
I have a feeling this number is a bit random (I'd be less concerned if it was a power of 2 like 8 or 16 or even 12 for that matter), but if it works I'm fine with it.